apps/admin/frontend/src/components/write_in_adjudication_button.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState } from 'react';
+import styled, { useTheme } from 'styled-components';
+import { CheckboxButton, Icons, SearchSelect } from '@votingworks/ui';
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+
+  /* Prevent the CheckboxButton bottom border from being cutoff on hover
+   * from the SearchSelect, which has a higher z-index */
+  button:focus-visible {
+    z-index: 10;
+  }
+`;
+
+export function WriteInAdjudicationButton({
+  caption,
+  isFocused,
+  isSelected,
+  hasInvalidEntry,
+  onChange,
+  onInputFocus,
+  onInputBlur,
+  toggleVote,
+  value,
+  officialCandidateNames,
+  writeInCandidateNames,
+}: {
+  caption?: React.ReactNode;
+  isFocused: boolean;
+  isSelected: boolean;
+  hasInvalidEntry: boolean;
+  // newVal can be name of new or existing candidate,
+  // keyword 'invalid' for invalid write-in, or undefined
```

Comment:
Consider using a different type to encode the fact that "Not a Mark" was chosen. Using strings with special meanings like this is a [well-known anti-pattern](https://www.theverge.com/tldr/2019/8/14/20805543/null-license-plate-california-parking-tickets-violations-void-programming-bug).